components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -169,15 +169,16 @@ impl DedicatedWorkerGlobalScope {
             };
 
             let runtime = Rc::new(ScriptTask::new_rt_and_cx());
+            let serialized_url = url.serialize();
             let global = DedicatedWorkerGlobalScope::new(
-                worker_url, id, devtools_chan, runtime.clone(), resource_task,
+                url, id, devtools_chan, runtime.clone(), resource_task,
```

Comment:
This change creates the worker scope using the post-redirect `final_url` instead of the pre-redirect `worker_url`.